File: app/src/main/java/com/chenchen/bee_rider/view/MyMapView.java
package com.chenchen.bee_rider.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import com.amap.api.maps.AMapOptions;
import com.amap.api.maps.TextureMapView;

/**
 * 创建人：进京赶考
 * 创建时间：2020/11/20  15：30
 * 描述：
 */
public class MyMapView extends TextureMapView {
    public MyMapView(Context context) {
        super(context);
    }

    public MyMapView(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }

    public MyMapView(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
    }

    public MyMapView(Context context, AMapOptions aMapOptions) {
        super(context, aMapOptions);
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        getParent().requestDisallowInterceptTouchEvent(true);
        return super.dispatchTouchEvent(ev);
    }
}

File: app/src/main/java/com/chenchen/bee_rider/view/MyMapView.kt
package com.chenchen.bee_rider.view

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import com.amap.api.maps.AMapOptions
import com.amap.api.maps.TextureMapView

/**
 * 创建人：进京赶考
 * 创建时间：2020/11/20  15：30
 * 描述：
 */
class MyMapView : TextureMapView {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attributeSet: AttributeSet?) : super(context, attributeSet) {}
    constructor(context: Context?, attributeSet: AttributeSet?, i: Int) : super(
        context,
        attributeSet,
        i
    ) {
    }

    constructor(context: Context?, aMapOptions: AMapOptions?) : super(context, aMapOptions) {}

    override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
        parent.requestDisallowInterceptTouchEvent(true)
        return super.dispatchTouchEvent(ev)
    }
}